wave_lang/kernel/wave/debug_log_hoist.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2025 The IREE Authors
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from .._support.tracing import CapturedTrace
+from ..lang.global_symbols import *
+from ..ops.wave_ops import (
+    get_custom,
+    Write,
+    Placeholder,
+)
+from .._support.dtype import DataType
+from .._support.indexing import IndexSymbol
+from typing import List, Tuple, TypedDict
```

Comment:
Nit: `List` and `Tuple` are deprecated since Python 3.9 https://docs.python.org/3/library/typing.html#typing.List. It looks like our support matrix starts with 3.10 so let's not use those maybe?